src/backend/functions/teams.ts

```diff
@@ -0,0 +1,117 @@
+/*
+ * This file is part of NER's PM Dashboard and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import { Handler } from '@netlify/functions';
+import { Description_Bullet, Prisma, PrismaClient, WBS_Element } from '@prisma/client';
+import {
+  ApiRoute,
+  ApiRouteFunction,
+  apiRoutes,
+  API_URL,
+  buildServerFailureResponse,
+  buildSuccessResponse,
+  routeMatcher,
+  Team,
+  WbsNumber,
+  DescriptionBullet
+} from 'utils';
+
+const prisma = new PrismaClient();
+
+const relationArgs = Prisma.validator<Prisma.TeamArgs>()({
+  include: {
+    members: true,
+    leader: true,
+    projects: {
+      include: {
+        wbsElement: true
+      }
+    }
+  }
+});
+
+const wbsNumOf = (element: WBS_Element): WbsNumber => ({
+  carNumber: element.carNumber,
+  projectNumber: element.projectNumber,
+  workPackageNumber: element.workPackageNumber
+});
+
+export const descBulletConverter = (descBullet: Description_Bullet): DescriptionBullet => ({
+  ...descBullet,
+  id: descBullet.descriptionId,
+  dateDeleted: descBullet.dateDeleted ?? undefined
+});
+
+const teamsTransformer = (team: Prisma.TeamGetPayload<typeof relationArgs>): Team => {
+  if (team === null) throw new TypeError('Team not found');
+
+  return {
+    teamId: team.teamId ?? undefined,
+    teamName: team.teamName ?? undefined,
+    leader: team.leader ?? undefined,
+    members: team.members ?? undefined,
+    projects: team.projects.map((project) => ({
+      id: project.projectId,
+      wbsNum: wbsNumOf(project.wbsElement),
+      name: project.wbsElement.name
+    }))
+  };
+};
+
+// Fetch all teams
+const getAllTeams: ApiRouteFunction = async () => {
+  const teams = await prisma.team.findMany(relationArgs);
+  return buildSuccessResponse(teams.map(teamsTransformer));
+};
+
+// make a team with the given leader and name
+const createTeam: ApiRouteFunction = async (params: { leaderId: string; teamName: string }) => {
+  const { teamName } = params;
```

Comment:
Does this work correctly for post body payload? I wanna say this looks like path parameters or query string parameters, but of which aren't the ideal implementation here

Reply:
true, gonna delete this for now just to get the basic stuff out